### PR TITLE
Add JavaScript Functional Programming

### PR DESCRIPTION
**Pull request by stoeffel**
This adds [Awesome FP JS](https://github.com/stoeffel/awesome-fp-js) to the list.
How do you like the list? Is there anything we have to change/add?

Thanks.

// cc @naehrstoff 
closes https://github.com/stoeffel/awesome-fp-js/issues/10

### CODE REVIEW

**Comment by sindresorhus** [comment]
Generally looks pretty good.

Some feedback:

- I would linkify `functional programming` in the description to a good source that explains functional programming. Maybe also add a succinct one-two sentences about functional programming at the top.
- `Inspired by the Awesome list thing.` You can remove this if you want. The badge is enough.
- The `People` sections isn't very useful as it gives no context about who they are and why I should care. They could use a bio and why I should follow them.
- The `Contribution` section needs a link to the contributing.md file.
- Many of the items in the `Articles` section could use a description/commentary.

**Comment by stoeffel** [comment]
Hey, thanks for the great feedback!
I will address it and bump you as soon as it's done.
Have nice Sunday and thanks for your time.

**Comment by sindresorhus** [comment]
ping :)

**Comment by stoeffel** [comment]
Sorry currently pretty busy. Will try to clean up this WE.
Thanks for the ping 😄

**Comment by stoeffel** [comment]
> The People sections isn't very useful as it gives no context about who they are and why I should care. They could use a bio and why I should follow them.

@naehrstoff how do you feel about removing this section? I do think this section is not very helpful atm and it would need a lot of time to keep it updated.

**Comment by stoeffel** [comment]
@sindresorhus: @naehrstoff and I did work on your feedback and think it's ready now. Would be great if you could have a second look. Please let me know if we should change/add something else.
Thanks for your time.

**Comment by sindresorhus** [comment]
Looks good. Thanks for working on improving it :)

> Maybe also add a succinct one-two sentences about functional programming at the top.

I'd still like to see this :arrow_up:

File: readme.md [hunk 1]
```diff
@@ -82,6 +82,8 @@ Check out my [blog](https://blog.sindresorhus.com) and follow me on [Twitter](ht
 	- [npm](https://github.com/sindresorhus/awesome-npm)
 	- [AVA](https://github.com/sindresorhus/awesome-ava) - Test runner
 	- [ESLint](https://github.com/dustinspecker/awesome-eslint)
+	- [AVA](https://github.com/sindresorhus/awesome-ava) *(Test runner)*
```
Comment (sindresorhus) [inline readme.md]: Always look at the diff. This got in. I've fixed it for you.

**Comment by naehrstoff** [comment]
I have a snippet of text that we can use as a short introduction to functional programming. Will add it soon-ish :)

File: readme.md (hunk 1 above)
Comment (stoeffel) [inline readme.md]: sorry for that.

**Comment by stoeffel** [comment]
Thanks @sindresorhus for merging and the great feedback!

PS: Sorry for the late "thank you". I was really busy 😊

**Pull request closed without merging**